index.src.html

```diff
@@ -586,33 +588,34 @@ <h2 id="monkeypatches">Patches to other specifications</h2>
 
 <h3 id="monkeypatch-fetch">Fetch</h3>
 
-The <a spec="FETCH">HTTP-network fetch</a> algorithm is modified by inserting a step after step 5, i.e. after request body streaming has begun but before response body streaming begins. This must call [=update the origin policy from a response=] given the response and the request's [=request/client=]. (This could cause the algorithm to block while it fetches an origin policy, if required do to so by the `<a http-header><code>Origin-Policy</code></a>` header.)
-
-<h3 id="monkeypatch-fp">Feature Policy</h3>
+Every [=response=] gets an associated <dfn for="response">origin policy</dfn>, which is an [=/origin policy=]. It is initially the [=null origin policy=].
 
-The [=create a feature policy for a browsing context=] algorithm is modified by modifying the returned [=feature policy=]'s <a spec="FEATURE-POLICY">declared policy</a> to be a [=map/clone=] of <var ignore>origin</var>'s [=origin/origin policy=]'s [=origin policy/feature policy=].
+The <a spec="FETCH">main fetch</a> algorithm is modified by inserting the following step after step 10. (This is after headers have been received and much of the other preliminaries processed, but the request body is likely still downloading.)
```

Comment:
This would mean it runs for non-HTTP responses as well, is that intended?

Reply:
Kind of. "get an origin policy for a response" will return the null origin policy with if the response's URL is non-HTTPS. This then lets us treat all responses as always having an origin policy, which is convenient.

However on second review I think this could move to HTTP fetch (or HTTP cache-or-network fetch??). Moving it all the way up here was necessary at some point during my refactoring (probably to be a peer of [step 11](https://fetch.spec.whatwg.org/#main-fetch) which does similar things for CSP). But it doesn't seem to be necessary now. Mainly because I said that by default a response's origin policy is the null origin policy anyway.

That said if I moved it into one of those two I'm not sure which one I'd choose, and I'm still a little confused as to whether I'd properly be affecting the inner response. Whereas with the current PR it seems more clearly correct.

I'll probably merge this as-is once GitHub stops 500ing but I'd really value continuing the conversation with more advice on the right way to integrate with Fetch, so please continue to teach me :).